node/lib/util/open.js

```diff
@@ -43,6 +43,17 @@ const SubmoduleUtil       = require("./submodule_util");
 const SubmoduleConfigUtil = require("./submodule_config_util");
 const SubmoduleFetcher    = require("./submodule_fetcher");
 
+/**
+ * @enum {SUB_OPEN_OPTION}
+ * Flags that describe wheter to open a submodule if it is part of a merge.
+ */
+const SUB_OPEN_OPTION = {
+    FORCE_OPEN    : 0, // non-bare repo and open submodule when needed
```

Comment:
"when needed" and "force" seem like an odd fit.  Also, do we actually want this ever?

Reply:
yes, we still need this option for at least two reasons:
1. `FORCE_OPEN` means "open the submodule when creating the `subRep` object". And creating the `subRepo` object is called in many places, not just the merge. For example the `subRepo` objects are needed in `reset`, `stash`, `commit` etc.  It is hard to test if half open submodules work for all those case. So to be safe let's allow fully opening in other commands.
2. Now the default option for normal merge is `ALLOW_BARE`, but in case there are bugs in its workflow, we can quickly switch back to old merge workflow with `FORCE_OPEN` instead of going through painful git-revert process.